io/zenoh-transport/src/unicast/auth_segment.rs

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use zenoh_result::ZResult;
+use zenoh_shm::{api::common::types::ProtocolID, posix_shm::array::ArrayInSHM};
+
+const AUTH_SEGMENT_PREFIX: &str = "auth";
+
+pub(crate) type AuthSegmentID = u32;
+pub(crate) type AuthChallenge = u64;
+
+#[derive(Debug)]
+pub struct AuthSegment {
+    array: ArrayInSHM<AuthSegmentID, AuthChallenge, usize>,
+}
+
+impl AuthSegment {
```

Comment:
With the current SHM system there is some leaking due to the shm segment never be closed.
How this new system will clean up all `AuthSegment`?

Reply:
`AuthSegment`, as any other zenoh SHM component, uses `posix_shm::array::ArrayInSHM` for managing segments. Previous SHM system used posix segment mapping on the filesystem - this mechanism required to manually delete SHM-mapped files to free shared memory and was unacceptable because our process might crash with no way to delete mapped files. The new SHM system uses system handles that are automatically unlinked when last linked process terminates, avoiding memory leakage.